### ISSUE TITLE
Build binaries for Windows ARM64

### ISSUE THREAD

**Issue by AArnott** [jellyfin/jellyfin-media-player#214]
Windows ARM64 devices are becoming more common. Running a media player under an x86 emulator consumes more battery than a natural arm64 build. Can you build and release arm64 binaries?

**Comment by iwalton3** [comment]
I don’t think GitHub Actions even supports ARM on Windows… I don’t really know where to start with this, but help would be appreciated

**Comment by AArnott** [comment]
You don't generally have to build on a Windows on ARM agent to build for ARM. 
If this were a .NET Core app (I guess it isn't), your build step just needs to be `dotnet build -r win-arm64`.
This is apparently a C++ project. At least the Microsoft C++ compiler and linker can target ARM from an x86/x64 agent. I'm not sure about linux build tools, but I'd guess they could.

**Comment by AArnott** [comment]
This media player crashed every few minutes when I install the x86 build on arm and run under emulation. I'd be happy to test the arm64 build if you come up with one.

**Comment by Betonhaus** [comment]
an ARM64 client would be appreciated

**Comment by sevenrats** [comment]
this isn't possible.
https://www.qt.io/blog/qt-for-windows-on-arm
@iwalton3 with your permission I think this should be closed as blocked by Qt, our unwillingness to build qt from source, and a lack of CI support.

**Comment by Betonhaus** [comment]
> this isn't possible. https://www.qt.io/blog/qt-for-windows-on-arm @iwalton3 with your permission I think this should be closed as blocked by Qt, our unwillingness to build qt from source, and a lack of CI support.

So if someone was to build Qt from source for this project it would be possible?

**Comment by Betonhaus** [comment]
A third party packaged Qt5 and Qt6 for WoA on MSYS2/MinGW.

https://packages.msys2.org/groups/mingw-w64-clang-aarch64-qt5

https://packages.msys2.org/groups/mingw-w64-clang-aarch64-qt6

Just Install MSYS2 and uses CLANGARM64 shell/environment

**Comment by minnyres** [comment]
> > this isn't possible. https://www.qt.io/blog/qt-for-windows-on-arm @iwalton3 with your permission I think this should be closed as blocked by Qt, our unwillingness to build qt from source, and a lack of CI support.
> 
> So if someone was to build Qt from source for this project it would be possible?

Qt is actually providing prebuilt binaries for WoA but QtWebengine is not included. QtWebengine cannot be built with MINGW. Build with MSVC is possible but not easy.

**Comment by MichelMelhem** [comment]
Hi, I have a surface laptop running Snapdragon x elite and it would be very nice to have an arm64 windows build :)

**Comment by theAeon** [comment]
Qt6.10 will have webengine built for WoA.

**Comment by theAeon** [comment]
Ironically, this means that this is also blocked by #599 unless someone hacks a qt5 build for WoA together, which sounds awful.

**Comment by grantbeattie** [comment]
The x64 build runs quite well on Windows/arm64 but a native build would be greatly appreciated 🙏

**Comment by AArnott** [comment]
Lacking an arm64 native client, I prefer to use the web client on arm64 devices because my browser at least is arm64 optimized.
The downside to this is it won't leverage my local hardware for transcoding... the server has to be enabled to do it.

**Comment by theAeon** [comment]
Just gave an ARM64 build a go for the fun of it.

The good news: it builds.

The bad news: it dies immediately.

Per WinDBG:

```
CreateWindow() for QEventDispatcherWin32 internal window failed (The operation completed successfully.)
(17f54.18bcc): Security check failure or stack buffer overrun - code c0000409 (!!! second chance !!!)
Subcode: 0x7 FAST_FAIL_FATAL_APP_EXIT 
```

and the message in particular: "failed to initialize mpv GL context"

Pulling down the qt debug syms to see if i can't get more info.

edit: wrong direction-it looks like the issue is that *mpv* doesn't want to initialize.

**Comment by theAeon** [comment]
https://github.com/jellyfin/jellyfin-desktop/blob/11cbc15697f7d807e7a6410e48675626249ea22c/src/main.cpp#L44-L46

I can't imagine that the windows.h include being behind an ifdef _M_X64 helps matters here.

(that said, fixing that doesn't actually help)

**Comment by theAeon** [comment]
https://github.com/jellyfin/jellyfin-desktop/blob/11cbc15697f7d807e7a6410e48675626249ea22c/src/system/SystemComponent.cpp#L146-L150

Also not great. Also irrelevant.

**Comment by talynone** [comment]
> Just gave an ARM64 build a go for the fun of it.
> 
> The good news: it builds.
> 
> The bad news: it dies immediately.
> 
> Per WinDBG:
> 
> ```
> CreateWindow() for QEventDispatcherWin32 internal window failed (The operation completed successfully.)
> (17f54.18bcc): Security check failure or stack buffer overrun - code c0000409 (!!! second chance !!!)
> Subcode: 0x7 FAST_FAIL_FATAL_APP_EXIT 
> ```
> 
> and the message in particular: "failed to initialize mpv GL context"
> 
> Pulling down the qt debug syms to see if i can't get more info.
> 
> edit: wrong direction-it looks like the issue is that _mpv_ doesn't want to initialize.

Hmm there is a WoA port of mpv:
https://github.com/mpv-player/mpv/releases

For hw decoding it must be started with correct command line argument:
'mpv.exe --hwdec'

**Comment by theAeon** [comment]
There's also an ARM build from the repo that this repo uses-which i used (and ran through the same steps). Currently trying to get it rebuilt with debug syms to figure out what the deal is.

**Comment by theAeon** [comment]
New error. GL's initializing fine at least (meaning the shinichiro/mpv-cmake-winbuild dlls are broken) but....yeah i don't think those functions are supposed to be at 0x0. Currently trying to trace the init of m_functions.


<img width="2304" height="1123" alt="Image" src="https://github.com/user-attachments/assets/bd3a2086-2d95-4be7-86df-345c002e1ebd" />

**Comment by theAeon** [comment]
<img width="1117" height="484" alt="Image" src="https://github.com/user-attachments/assets/a0347482-7780-4956-b1e5-ef9d46190bab" />

Well, that'd certainly explain why m_functions are null. Returning nullptr when threadcontext obviously exists makes no sense to me though (because the branch here indicates if(threadcontext) was false???)

**Comment by talynone** [comment]
> <img alt="Image" width="1117" height="484" src="https://private-user-images.githubusercontent.com/9099529/531088225-a0347482-7780-4956-b1e5-ef9d46190bab.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjcxNDY4MTgsIm5iZiI6MTc2NzE0NjUxOCwicGF0aCI6Ii85MDk5NTI5LzUzMTA4ODIyNS1hMDM0NzQ4Mi03NzgwLTQ5NTYtYjFlNS1lZjlkNDYxOTBiYWIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MTIzMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTEyMzFUMDIwMTU4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YzM3NmQyM2M3OWE4YTE1ZDc4OWM1MDc1NDMzNDcwMTY5Njc4ZDVlODQyY2EzNzg1N2Y5NWJlZWIzMzVhMWNkYyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.p6bVAjlbL9qX68mFdAEa3DlrXStjpv78MSnUDEYB-wQ">
> Well, that'd certainly explain why m_functions are null. Returning nullptr when threadcontext obviously exists makes no sense to me though (because the branch here indicates if(threadcontext) was false???)

https://chatgpt.com/share/695483f4-1830-8001-8491-fed7cf58cf1b

**Comment by theAeon** [comment]
<img width="1924" height="887" alt="Image" src="https://github.com/user-attachments/assets/b6fbd720-5642-43d9-8bc8-cab70c493ef2" />

it (kinda) works if i force software rendering-meaning its probably something to do with GLon12. Does anyone w/ a x86 system handy mind dropping the MS compatibility pack dlls alongside a 2.0 build and reporting back?

**Comment by talynone** [comment]
> Just a heads up - development of the Qt based Jellyfin Desktop has stopped in favor of promoting the SDL+CEF rewrite the new official desktop app.
> 
> There are currently arm64 Windows builds available, but I lack the means to test them. It's still early in the rewrite's development, but the x64 client seems pretty stable.
> 
> Dev builds can be found at https://github.com/jellyfin-labs/jellyfin-desktop-cef

Just tested the latest cd/cid ARM64 build, it crashes on startup. 

```
Faulting application name: jellyfin-desktop-cef.exe, version: 0.0.0.0, time stamp: 0x69b765d9
Faulting module name: jellyfin-desktop-cef.exe, version: 0.0.0.0, time stamp: 0x69b765d9
Exception code: 0xc0000409
Fault offset: 0x00000000002e678c
Faulting process id: 0xCE10
Faulting application start time: 0x1DCB572C79722AF
Faulting application path: C:\apps\jellyfindesktop\jellyfin-desktop-cef.exe
Faulting module path: C:\apps\jellyfindesktop\jellyfin-desktop-cef.exe
Report Id: 8de24239-47db-46be-91c7-6cd19903bbee
Faulting package full name: 
Faulting package-relative application ID: 
```

Personally I've switched to using my ARM64 fork of Fladder:, also with a mpv backend 
https://github.com/talynone/Fladder/releases

The upstream dev is waiting for Flutter to release to stable channel before merging in my changes:
https://github.com/DonutWare/Fladder/pull/747

**Comment by AArnott** [comment]
It crashed for me too. Only different was the fault offset:

```
Exception code: 0xc0000409
Fault offset: 0x00000000002e7f8c
```

**Comment by andrewrabert** [comment]
This Qt desktop client has been deprecated in favor of a rewrite: https://github.com/jellyfin/jellyfin-desktop
